main/test/ca/uwaterloo/flix/language/phase/TestResolver.scala

```diff
@@ -580,4 +580,131 @@ class TestResolver extends FunSuite with TestUtils {
     expectError[ResolutionError.UndefinedType](result)
   }
 
+  test("IllegalUninhabitedType.01") {
```

Comment:
What happens if you have:

```
def f(x: true)
```

or 

```
def f(x: {| x}
```

(Maybe the latter is allowed?)

Reply:
The bool case causes an error as expected; I'll add it to the test cases.

The record case I think should be allowed since `Record <:: Star`